qtpynodeeditor/examples/calculator.py

```diff
@@ -371,10 +373,10 @@ def main(app):
         registry.register_model(model, category='Operations',
                                 style=None)
 
-    registry.register_type_converter(DecimalData, IntegerData,
-                                     decimal_to_integer_converter)
-    registry.register_type_converter(IntegerData, DecimalData,
-                                     decimal_to_integer_converter)
+    dec_converter = TypeConverter(DecimalData.data_type, IntegerData.data_type, decimal_to_integer_converter)
```

Comment:
What do you think about allowing usage of the class itself for registering or performing conversions? 

I was thinking that class inheritance information could be used to indicate conversion compatibility, perhaps. At least in relatively simple graphs, it may even be possible to omit `data_type = NodeDataType(...)` then. That's aiming toward my hopeful overall goal of less boilerplate.

I'm definitely up for user feedback/input here, so don't take this as _"it must be this way"_.

~Although for backward compatibility, `TypeConverter(cls, cls, ...)` support should be left in for now, regardless. Depending on what we decide, it could be deprecated down the line.~ Oh, it looks like this was just a bug and wasn't supported to start with.

Reply:
> What do you think about allowing usage of the class itself for registering or performing conversions?

That would definitely reduce some boilerplate, i.e. I like it.

> I was thinking that class inheritance information could be used to indicate conversion compatibility

Perhaps, but there might be some very exotic conversions (e.g. an image into a text) and then the inheritance wouldn't help that much.

Reply:
> Perhaps, but there might be some very exotic conversions (e.g. an image into a text) and then the inheritance wouldn't help that much.

Agreed - which is likely where supporting the old `NodeDataType`-style type information would remain necessary. Alternatively, some other mechanism where you could opt out of it. More ideas for the future, in any case.